### PR TITLE
allow linking against the static msvc runtime library

### PR DESCRIPTION
**Pull request by ksophocleous**
the option `CARES_MSVC_STATIC_RUNTIME` only exists when configuring for Visual Studio compilers and it is `OFF` by default.

### CODE REVIEW

File: cmake/msvc_static_runtime.cmake [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FOREACH (flag_var
+    CMAKE_C_FLAGS CMAKE_C_FLAGS_DEBUG CMAKE_C_FLAGS_RELEASE
+    CMAKE_C_FLAGS_MINSIZEREL CMAKE_C_FLAGS_RELWITHDEBINFO)
+
+    IF (${flag_var} MATCHES "/MD")
+        STRING (REGEX REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
+    ENDIF (${flag_var} MATCHES "/MD")
```
Comment (gjasny) [inline cmake/msvc_static_runtime.cmake]: Please leave endif brackets empty.

File: cmake/msvc_static_runtime.cmake [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FOREACH (flag_var
+    CMAKE_C_FLAGS CMAKE_C_FLAGS_DEBUG CMAKE_C_FLAGS_RELEASE
```
Comment (gjasny) [inline cmake/msvc_static_runtime.cmake]: Instead of hardcoding configurations, could you please use CMAKE_CONFIGURATION_TYPES ?

Comment (ksophocleous) [inline cmake/msvc_static_runtime.cmake]: yes, very good idea.. done

File: cmake/msvc_static_runtime.cmake (hunk 1 above)
Comment (ksophocleous) [inline cmake/msvc_static_runtime.cmake]: right, also done

**Comment by ksophocleous** [comment]
@gjasny do you want me to squash these two commits?

**Comment by ksophocleous** [comment]
the change is done, do you want me to squash the commits?

**Pull request merged**